### PR TITLE
Support dynamic reprovisioning configuration in the IotEdgeSecurityDaemon.ps1 script

### PR DESCRIPTION
**Pull request by shantanu1singh**
Support dynamic reprovisioning configuration in the IotEdgeSecurityDaemon.ps1 script

### CODE REVIEW

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 1]
```diff
@@ -1744,14 +1751,22 @@ function Set-ProvisioningMode {
...
+        if ($DynamicReprovisioning) {
+            $DynamicReprovisioning = 'true'
+        }
+        else {
+            $DynamicReprovisioning = 'false'
+        }
+
         if ($ManualConnectionString -or $ManualX509) {
-            $selectionRegex = '(?:[^\S\n]*#[^\S\n]*)?provisioning:\s*#?\s*source:\s*".*"\s*#?\s*device_connection_string:\s*".*"'
+            $selectionRegex = '(?:[^\S\n]*#[^\S\n]*)?provisioning:\s*#?\s*source:\s*".*"\s*#?\s*device_connection_string:\s*".*"\s*#?\s*dynamic_reprovisioning:\s*".*"'
```
Comment (philipktlin) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: did ManualX509 support dynamic reprovisioning?

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 2]
```diff
@@ -1763,18 +1778,20 @@ function Set-ProvisioningMode {
...
         elseif ($External -or $ExternalProvisioningEndpoint){
-            $selectionRegex = '(?:[^\S\n]*#[^\S\n]*)?provisioning:\s*#?\s*source:\s*".*"\s*#?\s*endpoint:\s*".*"'
+            $selectionRegex = '(?:[^\S\n]*#[^\S\n]*)?provisioning:\s*#?\s*source:\s*".*"\s*#?\s*endpoint:\s*".*"\s*#?\s*dynamic_reprovisioning:\s*.*'
             $replacementContent = @(
                 'provisioning:',
                 '  source: ''external''',
-                "  endpoint: '$ExternalProvisioningEndpoint'")
+                "  endpoint: '$ExternalProvisioningEndpoint'",
+                "  dynamic_reprovisioning: $DynamicReprovisioning")
             $configurationYaml = ($configurationYaml -replace $selectionRegex, ($replacementContent -join "`n"))
```
Comment (philipktlin) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: Do we need to remove manual provisioning section like what you did in line 1835 in this PR?

Comment (shantanu1singh) [reply scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: Yes, you're right. Added that

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 (hunk 1 above)
Comment (shantanu1singh) [reply scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: Users have the option of enabling dynamic reprovisioning with the manual modes too. It's just that the daemon will stop execution if a reprovisioning scenario occurs and users will have to manually update the config file again.

This isn't really useful in any way but it was decided to just allow the config in all modes as it doesn't really make a difference

**Pull request merged**